### PR TITLE
Traitor tele implant

### PR DESCRIPTION
**Pull request by monster860**
This adds a new 5TC traitor item: the teleport implant.

Made from components salvaged from a stolen nuke disk, this implant will, upon use, teleport the user to a random location on the station.

:cl: monster860
rscadd: Adds the teleporter implant, a 5TC traitor item made from components salvaged from a nuke disk that teleports the user to a random location on-the station.
/:cl:

### CODE REVIEW

**Comment by Regen1** [comment]
:-1: 

This is lazily put inn and needs more work

It needs an emp_act, which should disable it or teleport them somewhere dangerous.
It needs some form of visual and text based message to anyone looking at the person teleporting, they shouldn't just vanish

It's a neat idea though.

**Comment by monster860** [comment]
Is this better?

**Comment by Regen1** [comment]
Looks good :+1:

**Comment by fludd12** [comment]
Maybe it's just an internalized "muh balance" meme, but I feel like it'd be pretty hilarious if leaving the Z-level snapped you back stationside, too, unless it was via emergency shuttle.

**Comment by Vivalas** [comment]
That would be pretty funny. This would turn into a fake-your-own-death-by-airlock / "lifeline" if you get spaced by validhunting assistants.

**Comment by Kyep** [comment]
I'm conflicted about this.

On the one hand, it is quite helpful to traitors, enabling them to escape custody, including perma, much more easily. As well as simply not being caught so easily in the first place. All of this is good, as it means they can continue making the round interesting for longer. So, I feel like I should be in favor of it.

That said, of the 24 blob spawn locations, 11, almost half, require access/tools/welder to get out of. So, a traitor using one of these has a good chance to trap themselves somewhere they cannot escape without tools. I'm also not sure what this will do to the way that sec play.

So overall: sounds interesting, but I am not sure how it would play out in practice.

**Comment by Belario** [comment]
@Kyep The way I see it, if a traitor knowingly uses a teleport implant set to random and they don't bring along hacking tools they deserve to locked in.

**Comment by Kyep** [comment]
Or use their uplink to buy a syndie toolbox.

**Comment by monster860** [comment]
@Kyep 

>I'm also not sure what this will do to the way that sec play.

The new meta will be to place a glass cage and a motion alarm camera at every blobstart.

File: code/game/objects/items/weapons/implants/implant_misc.dm [hunk 1]
```diff
@@ -63,4 +63,36 @@
...
+				<b>Implant Details:</b> Made from components salvaged from a nuke disk<BR>
+				<b>Function:</b> Teleports user to a random location<BR>
+				<b>Integrity:</b> Implant can only be used three times before reserves are depleted."}
+	return dat
+
+/obj/item/weapon/implant/teleport/activate()
+	if(uses < 1)
+		return 0
+	uses--
+	imp_in.visible_message("<span class='warning'>[imp_in] disappears in a flash of light!</span>", "<span class='warning'>You activate \the [src], teleporting you to a random location</span>")
```
Comment (TheDZD) [inline code/game/objects/items/weapons/implants/implant_misc.dm]: You don't need \the

File: code/game/objects/items/weapons/implants/implant_misc.dm [hunk 2]
```diff
@@ -63,4 +63,36 @@
...
+	imp_in.forceMove(get_turf(pick(blobstart)))
+
+/obj/item/weapon/implant/teleport/emp_act(severity)
+	if(uses < 1)
+		return
+	uses--
+	imp_in.visible_message("<span class='warning'>[imp_in] disappears in a flash of light!</span>", "<span class='warning'>\The [src] malfunctions, teleporting you!</span>")
```
Comment (TheDZD) [inline code/game/objects/items/weapons/implants/implant_misc.dm]: Here as well.

File: code/game/objects/items/weapons/implants/implant_misc.dm (hunk 1 above)
Comment (monster860) [inline code/game/objects/items/weapons/implants/implant_misc.dm]: It may not be necessary, but it makes it clear that there will be a the there.

Comment (tigercat2000) [inline code/game/objects/items/weapons/implants/implant_misc.dm]: you can argue the same for `src.`
`src. means it's more clear that it's calling a proc on itself`
still unnecessary code :V

**Comment by Fox-McCloud** [comment]
Closing; stale. No updates after multiple attempts on and off Git to update.

**Pull request closed without merging**